Signal/src/ViewControllers/ContactViewController.swift

```diff
@@ -0,0 +1,686 @@
+        label.autoPinTrailingToSuperviewMargin(withInset: 20)
+        label.autoVCenterInSuperview()
+        label.autoPinEdge(toSuperviewEdge: .top, withInset: 0, relation: .greaterThanOrEqual)
+        label.autoPinEdge(toSuperviewEdge: .bottom, withInset: 0, relation: .greaterThanOrEqual)
+
+        return button
+    }
+
+    func didPressCreateNewContact(sender: UIGestureRecognizer) {
+        Logger.info("\(self.TAG) \(#function)")
+
+        guard sender.state == .recognized else {
+            return
+        }
+        presentNewContactView()
+    }
+
+    func didPressAddToExistingContact(sender: UIGestureRecognizer) {
+        Logger.info("\(self.TAG) \(#function)")
+
+        guard sender.state == .recognized else {
+            return
+        }
+        presentSelectAddToExistingContactView()
+    }
+
+    func didPressShareContact(sender: UIGestureRecognizer) {
+        Logger.info("\(self.TAG) \(#function)")
+
+        guard sender.state == .recognized else {
+            return
+        }
+        // TODO:
+    }
+
+    func didPressSendMessage() {
+        Logger.info("\(self.TAG) \(#function)")
+
+        // TODO: We're taking the first Signal account id. We might
+        // want to let the user select if there's more than one.
+        guard let recipientId = systemContactsWithSignalAccountsForContact().first else {
+            owsFail("\(logTag) missing Signal recipient id.")
+            return
+        }
+
+        SignalApp.shared().presentConversation(forRecipientId: recipientId, action: .compose)
+    }
+
+    func didPressAudioCall() {
+        Logger.info("\(self.TAG) \(#function)")
+
+        // TODO: We're taking the first Signal account id. We might
+        // want to let the user select if there's more than one.
+        guard let recipientId = systemContactsWithSignalAccountsForContact().first else {
+            owsFail("\(logTag) missing Signal recipient id.")
+            return
+        }
+
+        SignalApp.shared().presentConversation(forRecipientId: recipientId, action: .audioCall)
+    }
+
+    func didPressVideoCall() {
+        Logger.info("\(self.TAG) \(#function)")
+
+        // TODO: We're taking the first Signal account id. We might
+        // want to let the user select if there's more than one.
+        guard let recipientId = systemContactsWithSignalAccountsForContact().first else {
+            owsFail("\(logTag) missing Signal recipient id.")
+            return
+        }
+
+        SignalApp.shared().presentConversation(forRecipientId: recipientId, action: .videoCall)
+    }
+
+    func didPressInvite() {
+        Logger.info("\(self.TAG) \(#function)")
+
+        guard MFMessageComposeViewController.canSendText() else {
+            Logger.info("\(TAG) Device cannot send text")
+            OWSAlerts.showErrorAlert(message: NSLocalizedString("UNSUPPORTED_FEATURE_ERROR", comment: ""))
+            return
+        }
+        let phoneNumbers =
+        phoneNumbersForContact()
```

Comment:
format

Reply:
clang-format is really disappointing me lately.